funding/manager.go

```diff
@@ -350,8 +350,10 @@ type Config struct {
 	NotifyWhenOnline func(peer [33]byte, peerChan chan<- lnpeer.Peer)
 
 	// FindChannel queries the database for the channel with the given
-	// channel ID.
-	FindChannel func(chanID lnwire.ChannelID) (*channeldb.OpenChannel, error)
+	// channel ID. Providing the node's public key is an optimization that
+	// prevents deserializing and scanning through all possible channels.
+	FindChannel func(node *btcec.PublicKey,
```

Comment:
Non-blocking, but could abstract this somewhat to a sort of abstract "locator" type depending on the implementation.

Reply:
what else would it locate?